src/core/meshcop/dataset.hpp

```diff
@@ -203,17 +203,23 @@ class Dataset
          * @returns The Active Timestamp in the Dataset.
          *
          */
-        uint64_t GetActiveTimestamp(void) const { return mActiveTimestamp; }
+        Timestamp GetActiveTimestamp(void) const
+        {
+            Timestamp timestamp;
+
+            timestamp.SetFromTimestamp(mActiveTimestamp);
+            return timestamp;
+        }
 
         /**
          * This method sets the Active Timestamp in the Dataset.
          *
          * @param[in] aTimestamp   A Timestamp value.
          *
          */
-        void SetActiveTimestamp(uint64_t aTimestamp)
+        void SetActiveTimestamp(Timestamp aTimestamp)
```

Comment:
I expect passing `Timestamp aTimetstamp` will increase code size (and very likely execution time) since we can potentially create a copy every time the method is called. I cannot think of any benefit for it over passing a `const` reference.

Reply:
Regarding performance, `Timestamp` has 64 bits as its non-static data member, so copying it is the same as copying an int64. Without LTO, I expect const reference performs better on 32 bit machines, and they perform the same on 64 bit machines.

C++ style guide doesn't have a strong preference. Generally it's better to pass by value if the type is small (<= 16 bytes) and trivially copyable. But I think the standard of being small varies case by case. We can consider <= 4 bytes or <= 8 bytes as OT is usually running on resource restricted environments.

Reply:
it turns out passing by const reference will reduce code size:
https://github.com/openthread/openthread/pull/7739/commits/a08c9c5715213c644cba134151c42291d19f75ba
```
  ot-cli-ftd_1.2               -1de969d  482664  780   70296  553740
                               +a08c9c5  482648  780   70296  553724
                               +/-       -16     0     0      -16
  ot-ncp-ftd_1.2               -1de969d  464180  780   69328  534288
                               +a08c9c5  464148  780   69328  534256
                               +/-       -32     0     0      -32
```

Maybe because the cpu size is 32 bits, so copying uint64_t has more overhead.